src/search/search_encoding.h

```diff
@@ -291,6 +359,92 @@ struct NumericFieldMetadata : IndexFieldMetadata {
   bool IsSortable() const override { return true; }
 };
 
+struct HnswVectorFieldMetadata : IndexFieldMetadata {
+  VectorType vector_type;
+  uint16_t dim;
+  DistanceMetric distance_metric;
+
+  uint32_t initial_cap = 500000;
+  uint16_t m = 16;
+  uint32_t ef_construction = 200;
+  uint32_t ef_runtime = 10;
+  double epsilon = 0.01;
+  uint16_t num_levels = 0;
+
+  HnswVectorFieldMetadata() : IndexFieldMetadata(IndexFieldType::VECTOR) {}
+
+  void Encode(std::string *dst) const override {
+    IndexFieldMetadata::Encode(dst);
+    PutFixed8(dst, uint8_t(vector_type));
+    PutFixed16(dst, dim);
+    PutFixed8(dst, uint8_t(distance_metric));
+    PutFixed32(dst, initial_cap);
+    PutFixed16(dst, m);
+    PutFixed32(dst, ef_construction);
+    PutFixed32(dst, ef_runtime);
+    PutDouble(dst, epsilon);
+    PutFixed16(dst, num_levels);
+  }
+
+  rocksdb::Status Decode(Slice *input) override {
+    if (auto s = IndexFieldMetadata::Decode(input); !s.ok()) {
+      return s;
+    }
+
+    if (input->size() < 1 + 2 + 1 + 4 + 2 + 4 + 4 + 8 + 2) {
+      return rocksdb::Status::Corruption(kErrorInsufficientLength);
+    }
+
+    GetFixed8(input, (uint8_t *)(&vector_type));
+    GetFixed16(input, &dim);
+    GetFixed8(input, (uint8_t *)(&distance_metric));
+    GetFixed32(input, &initial_cap);
+    GetFixed16(input, &m);
+    GetFixed32(input, &ef_construction);
+    GetFixed32(input, &ef_runtime);
+    GetDouble(input, (double *)(&epsilon));
```

Comment:
```suggestion
    GetDouble(input, &epsilon);
```

Do we need to cast it here?

Reply:
Great catch! Changed :)